pandas/tests/copy_view/test_functions.py

```diff
@@ -310,3 +310,93 @@ def test_merge_copy_keyword(using_copy_on_write, copy):
     else:
         assert not np.shares_memory(get_array(df, "a"), get_array(result, "a"))
         assert not np.shares_memory(get_array(df2, "b"), get_array(result, "b"))
+
+
+def test_join_on_key(using_copy_on_write):
+    """Test if DataFrame.join applies Copy-On-Write optimization.
+
+    GIVEN two DataFrame instances
+    WHEN DataFrame.join is called for one of them
+    THEN check that the result DataFrame instance
+        shares the same memory with original dataframes until it is edited.
+    """
+    df1 = DataFrame({"key": ["a", "b", "c"], "a": [1, 2, 3]})
+    df2 = DataFrame({"key": ["a", "b", "c"], "b": [4, 5, 6]})
```

Comment:
Can you define ``index=Index(["a", "b", "c"], name="key")`` instead of using it as a column? We always try to create a test with the least number of operations possible

Reply:
Sure, thank you for the review. Implemented changes based on the comments.